sdk/storage/azure-storage-blobs/src/blob_client.cpp

```diff
@@ -198,12 +188,12 @@ namespace Azure { namespace Storage { namespace Blobs {
         unused(context);
 
         DownloadBlobOptions newOptions = options;
-        newOptions.Offset
-            = (options.Offset.HasValue() ? options.Offset.GetValue() : 0) + retryInfo.Offset;
-        newOptions.Length = options.Length;
-        if (newOptions.Length.HasValue())
+        newOptions.Range = Core::Http::Range();
+        newOptions.Range.GetValue().Offset
+            = (options.Range.HasValue() ? options.Range.GetValue().Offset : 0) + retryInfo.Offset;
+        if (options.Range.HasValue() && options.Range.GetValue().Length.HasValue())
         {
-          newOptions.Length = options.Length.GetValue() - retryInfo.Offset;
+          newOptions.Range.GetValue().Length = options.Range.GetValue().Length.GetValue() - retryInfo.Offset;
```

Comment:
note to self: see if there is a better way to write this rather than repeated checks to `.Range.HasValue()`.

Do we need an `Http::Range()` ctor that takes offset and length to make creating such range creation easier? 
Could a pattern like `options.Range = {1_MB, 2_MB};` be applied here?

Reply:
> Do we need an Http::Range() ctor that takes offset and length to make creating such range creation easier?

I'm not opposed to adding such constructor(s), although I don't think it's necessary. But what I want to say is, the `Length` of an `Http::Range` can be nullable. If you want to add a constructor that instantiates a range with a specific length, you might also want to add a constructor that instantiates a range without length.

Reply:
> Could a pattern like options.Range = {1_MB, 2_MB}; be applied here?

This is [aggregate initialization](https://en.cppreference.com/w/cpp/language/aggregate_initialization), which is not allowed. We talked about this in one of the guideline meetings.